icu4c/source/test/cintltst/cstrcase.c

```diff
@@ -748,9 +748,9 @@ TestUCaseMap(void) {
     /* overly long locale IDs may get truncated to their language code to avoid unnecessary allocation */
     ucasemap_setLocale(csm, "I-kLInGOn-the-quick-brown-fox-jumps-over-the-lazy-dog", &errorCode);
     locale=ucasemap_getLocale(csm);
-    if(0!=strncmp(locale, "tlh", 3)) {
+    if(0!=strncmp(locale, "the", 3)) {
```

Comment:
I see... it maps i-klingon to tlh but keeps the rest of the tag, and because the next subtag is "the" which fits the extlang syntax it replaces the language subtag...

For now, I guess we call it "garbage in, garbage out".

Please
1. add a comment here for why we get "the"
2. submit a follow-up ticket to revisit this and figure out if we want to tweak the extlang handling

Reply:
ok added comments
and
https://unicode-org.atlassian.net/browse/ICU-20432